src/everwillow/inference/uncertainty.py

```
@@ -0,0 +1,176 @@
+from jaxtyping import Array, Float, PyTree
+
+import everwillow.statelib as sl
+from everwillow.statelib import V
+
+
+def hessian_matrix(
+    nll_fn: tp.Callable[[PyTree[V]], float],
+    params: sl.State[V],
+    *,
+    fixed: sl.State[V | EllipsisType] | None = None,
+) -> Float[Array, "n_free n_free"]:
```

Ruff (0.14.13)

Syntax error in forward annotation: Unexpected token at the end of an expression

(F722)